pkg/provisioner/ironic/ironic.go

```diff
@@ -1196,14 +1261,9 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 			nodes.ProvisionStateOpts{Target: nodes.TargetProvide})
 
 	case nodes.CleanFail:
-		if ironicNode.Maintenance {
-			p.log.Info("clearing maintenance flag")
-			return p.setMaintenanceFlag(ironicNode, false)
-		}
-		return p.changeNodeProvisionState(
-			ironicNode,
-			nodes.ProvisionStateOpts{Target: nodes.TargetManage},
-		)
+		result.Dirty = true
+		result.ErrorMessage = "clean failed"
```

Comment:
I'm not sure how I feel about relying on this implicit string constant to distinguish between error types. I under go does not have typed errors (sigh), but can we maybe have a flag or at least make this string a constant?

Reply:
Agree, and also would make sense to keep the management of _manual_ and _automated_ cleaning failure separate (instead of going back to Preparing in case of automated clean failure)?